src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -244,6 +244,12 @@ impl<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 return;
             }
 
+            PassMode::Ignore(IgnoreMode::Uninhabited) => {
```

Comment:
Why not check `self.fn_ty.ret.layout.abi.is_uninhabited()`?

I haven't yet reviewed the PR that added `IgnoreMode` but I suspect it's not the best solution to the problem.

Reply:
The way I have it now, `librustc_codegen_llvm` controls both the addition of `noreturn` and setting the mode to `PassMode::Ignore(IgnoreMode::Uninhabited)`. That seemed like the right way, so llvm/ssa don't mix responsibilities too much.

Are you suggesting to leave the pass mode entirely alone, and just check uninhabited -> abort here? I can try that...